source/mge.js

```diff
@@ -0,0 +1,988 @@
+                if (type === "list") {
+                    let content = "";
+                    for (const child of this.children_defs) {
+                        content += child.formatString() + ',';
+                    }
+                    return "[" + content + "]";
+                }
+                if (type === "dict") {
+                    var content = "";
+                    for (var i = 0; i < this.children_defs.length; i++) {
+                        content += this.aux_data[i] + ":" + this.children_defs[i].formatString;
+                    }
+                    return "{" + content + "}";
+                }
+                return "unknown";
+            }
+        });
+        this.registerType('megengine.traced_module.pytree.LeafDef', class {
+            toString() {
+                let content = '';
+                if (this.const_val !== null) {content += this.const_val;}
+                else {content += '[';}
+                for (var t of Object.values(this.type)) {
+                    content += t.__name__;
+                }
+                content += ']';
+                return content;
+            }
+
+            formatString() {
+                if (this.const_val !== null) {return this.const_val;}
+                else
+                if (this.type[0].__module__ !== undefined) {return this.type[0].__name__;}
+                return "None";
+            }
+        });
+        this.registerType('megengine.tensor.Tensor', class {
+            constructor(data, dtype, device) {
+                this.data = data;
+                this.dtype = dtype;
+                this.device = device;
+            }
+        });
+
+        this.registerType('megengine.core.tensor.dtype.QuantDtypeMeta', class {
+            constructor(name, cname, np_dtype, qmin, qmax, is_signed) {
+                this.name = name;
+                this.cname = cname;
+                this.np_dtype = np_dtype;
+                this.qmin = qmin;
+                this.qmax = qmax;
+                this.is_signed = is_signed;
+            }
+        });
+
+    }
+};
+
+megengine.Container = class {
+
+    static open(context) {
+
+        const pickle = megengine.Container.Pickle.open(context.stream);
+        if (pickle) {
+            return pickle;
+        }
+
+        return null;
+    }
+};
+
+
+megengine.Container.Pickle = class {
+
+    static open(stream) {
+        return new megengine.Container.Pickle(stream);
+    }
+
+    constructor(stream) {
+        this._stream = stream;
+        this._graphs = [this];
+    }
+
+    set metadata(value) {
+        this._metadata = value;
+    }
+
+    set exception(value) {
+        this._exceptionCallback = value;
+    }
+
+    get format() {
+        this._unpickle();
+        var version = "1.6";
+        var dump_info = this._graphs[0].data.dump_info;
+        if (dump_info !== undefined) {
+            version = dump_info.version;
+        }
+        return 'megengine v' + version;
```

Comment:
`MegEngine` to use same casing as repo and Pip package?

Reply:
the name of Pip package is `MegEngine`,  and I modified the format to `MegEngine + version`